pkg/api/util.go

```diff
@@ -211,13 +211,54 @@ func mapStructure(input, output any, hooks map[string]func(v string) (string, er
 			field.SetString(value)
 		case reflect.Slice:
 			if value == "" {
-				return nil // Nil slice.
+				return nil
 			}
-			val, err := hex.DecodeString(value)
-			if err != nil {
-				return err
+
+			elemT := field.Type().Elem()
+
+			// 1) []byte — (hex → []byte)
+			if elemT.Kind() == reflect.Uint8 {
+				val, err := hex.DecodeString(value)
+				if err != nil {
+					return err
+				}
+				field.SetBytes(val)
+				return nil
+			}
+
+			// 2) []byte -> []multiaddr.Multiaddr
+			if elemT.Kind() == reflect.Interface &&
+				elemT == reflect.TypeOf((*multiaddr.Multiaddr)(nil)).Elem() {
+
+				parts := strings.Split(value, ",")
+				out := reflect.MakeSlice(field.Type(), 0, len(parts))
+				seen := make(map[string]struct{}, len(parts))
```

Comment:
Why do we have any changes in mapStructure? It is only used in api handlers and I didn't found any handler that is using as an input slice of Multiaddr?